File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    for i in s:
        if len(stack) == 0:
            stack.append(i)
        elif i == ")" and stack[-1] == "(":
            stack.pop()
        else:
            stack.append(i)
    if len(stack) == 0:
        answer = True
    else:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (int i=0;i<s.length();i++) {
            char c = s.charAt(i);
            if (c == '(') {
                stack.push(c);
            } else {
                if (stack.size() == 0) {
                    stack.push(c);
                }
                else if (stack.peek() == '(') {
                    stack.pop();
                }
            }
        }
        if (stack.size() == 0) {
            return true;
        } else {
            return false;
        }
    }
}